control/lti.py

```diff
@@ -451,32 +505,32 @@ def evalfr(sys, x):
 
     .. todo:: Add example with MIMO system
     """
-    if issiso(sys):
-        return sys.horner(x)[0][0]
-    return sys.horner(x)
-
+    return sys.__call__(x, squeeze=squeeze)
 
-def freqresp(sys, omega):
+def freqresp(sys, omega, squeeze=True):
     """
     Frequency response of an LTI system at multiple angular frequencies.
 
     Parameters
     ----------
     sys: StateSpace or TransferFunction
         Linear system
-    omega: array_like
+    omega: float or array_like
         A list of frequencies in radians/sec at which the system should be
         evaluated. The list can be either a python list or a numpy array
         and will be sorted before evaluation.
+    squeeze: bool, optional (default=True)
+        If True and sys is single input, single output (SISO), returns
+        1D array or scalar depending on omega's length.
```

Comment:
```suggestion
    squeeze : bool, optional (default=True)
        If True and `sys` is single input, single output (SISO), returns
        1D array instead of 3D array.
```